Sources/Sentry/SentrySDK.m

```diff
@@ -134,24 +135,26 @@ + (void)setStartInvocations:(NSUInteger)value
 
 + (void)startWithOptions:(SentryOptions *)options
 {
-    startInvocations++;
+    [SentryDependencyContainer.sharedInstance.dispatchQueueWrapper dispatchSyncOnMainQueue:^{
+        startInvocations++;
 
-    [SentryLog configure:options.debug diagnosticLevel:options.diagnosticLevel];
+        [SentryLog configure:options.debug diagnosticLevel:options.diagnosticLevel];
 
-    SentryClient *newClient = [[SentryClient alloc] initWithOptions:options];
-    [newClient.fileManager moveAppStateToPreviousAppState];
-    [newClient.fileManager moveBreadcrumbsToPreviousBreadcrumbs];
+        SentryClient *newClient = [[SentryClient alloc] initWithOptions:options];
+        [newClient.fileManager moveAppStateToPreviousAppState];
+        [newClient.fileManager moveBreadcrumbsToPreviousBreadcrumbs];
```

Comment:
I worry that dispatching this synchronously to the main thread creates any potential for deadlocks like have been reported in https://github.com/getsentry/sentry-cocoa/issues/3277.

Any reason it can't be done asynchronously?

Reply:
Basically, everything that we use from UIKit should be used in the main thread.
Also, it's good to make sure the SDK is initialized before the app start doing its thing, ensuring we can capture more events.

Regarding #3277, we need to do some refactoring to prevent that from happening.